doc/kernel/services/signals.rst

```diff
@@ -0,0 +1,204 @@
+.. _signals:
+
+Signal API
+##########
+
+The signal API supports signalling between threads.
+
+.. contents::
+    :local:
+    :depth: 2
+
+Concepts
+********
+
+The signal API provides a way for one thread to notify another thread that an
+event has occurred using a predefined integer value (see signal numbers).
+
+Zephyr's signal API can be broken down into functions that manipulate and
+query *signal sets* (:c:struct:`k_sig_set`), and functions that are used to
+send and receive signals (:c:func::).
+
+Standard vs Real-time Signals
+=============================
+
+Signals may be divided into two categories; *standard* and *real-time*. The
+main differences between the two are
+
+* ISO C and POSIX do not specify whether standard signals are queued or merged.
+* ISO C and POSIX do not specify the priority of standard signals.
+
+To illustrate the former point, if ``SIGALRM`` is sent twice to an
+application, the implementation may merge those into one signal or queue them.
+On the other hand, Real-time signals are always queued in the order they are
+generated, and are never merged.
+
+Standard signals therefore have a shortcoming at the specification level; there
+can be no guarantee of determinism if a signal generated at one time instance
+is merged with another signal generated at another time instance. Zephyr removes
+this ambiguity by delivering all signals in the order they are generated, and
+not merging standard signals. Zephyr's handling of standard signals is similar
+to how Zephyr handles real-time signals.
+
+.. note::
+    Zephyr queues all signals in the order they are generated, and never
+    merges any signals (real-time or otherwise).
+
+To address the latter point, real-time signals have a priority with respect to
+other real-time signals. However, the POSIX specification does not clarify the
```

Comment:
is this supposed to be:
```suggestion
other standard signals. However, the POSIX specification does not clarify the
```

Reply:
No, POSIX only specifies priority between real-time signals. There is no specification that standard signals have higher or lower priority than real-time signals.

I think it's better to be unambiguous though and clearly make the distinction that real-time signals have priority over standard signals.